core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -96,6 +114,7 @@ private[spark] object JsonProtocol {
         executorMetricsUpdateToJson(metricsUpdate)
       case blockUpdated: SparkListenerBlockUpdated =>
         throw new MatchError(blockUpdated)  // TODO(ekl) implement this
+      case _ => parse(write(event))
```

Comment:
Hi @steveloughran, I think currently the generic handling is only for new events added in the sub modules.